File: web/src/api/miser/miser_ranking_record.js
import service from '@/utils/request'

export const createMiserRankingRecord = (data) => {
  return service({
    url: '/miserRankingRecord/createMiserRankingRecord',
    method: 'post',
    data
  })
}

export const deleteMiserRankingRecord = (params) => {
  return service({
    url: '/miserRankingRecord/deleteMiserRankingRecord',
    method: 'delete',
    params
  })
}

export const deleteMiserRankingRecordByIds = (params) => {
  return service({
    url: '/miserRankingRecord/deleteMiserRankingRecordByIds',
    method: 'delete',
    params
  })
}

export const updateMiserRankingRecord = (data) => {
  return service({
    url: '/miserRankingRecord/updateMiserRankingRecord',
    method: 'put',
    data
  })
}

export const findMiserRankingRecord = (params) => {
  return service({
    url: '/miserRankingRecord/findMiserRankingRecord',
    method: 'get',
    params
  })
}

export const getMiserRankingRecordList = (params) => {
  return service({
    url: '/miserRankingRecord/getMiserRankingRecordList',
    method: 'get',
    params
  })
}

File: server/router/miser/miser_ranking_record.go
package miser

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type MiserRankingRecordRouter struct{}

func (s *MiserRankingRecordRouter) InitMiserRankingRecordRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	miserRankingRecordRouter := Router.Group("miserRankingRecord").Use(middleware.OperationRecord())
	{
		miserRankingRecordRouter.POST("createMiserRankingRecord", miserRankingRecordApi.CreateMiserRankingRecord)
		miserRankingRecordRouter.DELETE("deleteMiserRankingRecord", miserRankingRecordApi.DeleteMiserRankingRecord)
		miserRankingRecordRouter.DELETE("deleteMiserRankingRecordByIds", miserRankingRecordApi.DeleteMiserRankingRecordByIds)
		miserRankingRecordRouter.PUT("updateMiserRankingRecord", miserRankingRecordApi.UpdateMiserRankingRecord)
	}

	miserRankingRecordRouterWithoutRecord := Router.Group("miserRankingRecord")
	{
		miserRankingRecordRouterWithoutRecord.GET("findMiserRankingRecord", miserRankingRecordApi.FindMiserRankingRecord)
		miserRankingRecordRouterWithoutRecord.GET("getMiserRankingRecordList", miserRankingRecordApi.GetMiserRankingRecordList)
	}

	miserRankingRecordRouterWithoutAuth := PublicRouter.Group("miserRankingRecord")
	{
		_ = miserRankingRecordRouterWithoutAuth
	}
}
